test/functional/autocmd/autocmd_spec.lua

```diff
@@ -367,26 +367,26 @@ describe('autocmd', function()
     -- This used to crash after making aucmd_win a normal window via the above.
     exec_lua [[
       vim.cmd "tabnew | tabclose # | wincmd s | wincmd _"
-      vim.api.nvim_buf_call(_G.buf, function() vim.cmd "wincmd K" end)
+      vim._with({buf = _G.buf}, function() vim.cmd "wincmd K" end)
     ]]
     assert_alive()
     screen:expect_unchanged()
 
     -- Also check with win_splitmove().
     exec_lua [[
-      vim.api.nvim_buf_call(_G.buf, function()
+      vim._with({buf = _G.buf}, function()
         vim.fn.win_splitmove(vim.fn.winnr(), vim.fn.win_getid(1))
       end)
     ]]
     screen:expect_unchanged()
 
     -- Also check with nvim_win_set_config().
     matches(
-      ': Failed to move window %d+ into split$',
+      '^Failed to move window %d+ into split$',
```

Comment:
Hmm... I wonder why this test changes was necessary? On paper `vim._with({ buf = buf }, f)` seems to work exactly as `vim.api.nvim_buf_call(buf, f)`.

Reply:
Just using different error functions I think. `vim._with` uses `nlua_push_errstr` while `nvim_buf_call` uses `api_set_error`. I don't think it makes a real difference which you use.